collect_app/src/test/java/org/odk/collect/android/widgets/utilities/GeoWidgetUtilsTest.kt

```diff
@@ -0,0 +1,171 @@
+        )
+        assertEquals(
+            "W 150°55'37\"",
+            convertCoordinatesIntoDegreeFormat(context, -150.92708666666667, "lon")
+        )
+        assertEquals("N 0°0'0\"", convertCoordinatesIntoDegreeFormat(context, 0.0, "lat"))
+        assertEquals("E 0°0'0\"", convertCoordinatesIntoDegreeFormat(context, 0.0, "lon"))
+    }
+
+    @Test
+    fun floorTest() {
+        assertEquals("5", floor("5"))
+        assertEquals("-5", floor("-5"))
+        assertEquals("5", floor("5.55"))
+        assertEquals("-5", floor("-5.55"))
+        assertEquals("", floor(""))
+        assertEquals("", floor(null))
+        assertEquals("qwerty", floor("qwerty"))
+    }
+
+    @Test
+    fun parseGeometryPointTest() {
+        var gp =
+            parseGeometryPoint("37.45153333333334 -122.15539166666667 0.0 20.0")!!
+        assertEquals(37.45153333333334, gp[0])
+        assertEquals(-122.15539166666667, gp[1])
+        assertEquals(0.0, gp[2])
+        assertEquals(20.0, gp[3])
+
+        gp = parseGeometryPoint("37.45153333333334")!!
+        assertEquals(37.45153333333334, gp[0])
+        assertEquals(0.0, gp[1])
+        assertEquals(0.0, gp[2])
+        assertEquals(0.0, gp[3])
+
+        gp = parseGeometryPoint(" 37.45153333333334 -122.15539166666667 0.0 ")!!
+        assertEquals(37.45153333333334, gp[0])
+        assertEquals(-122.15539166666667, gp[1])
+        assertEquals(0.0, gp[2])
+        assertEquals(0.0, gp[3])
+
+        assertEquals(null, parseGeometryPoint("37.45153333333334 -122.15539166666667 0.0 qwerty"))
+        assertEquals(null, parseGeometryPoint(""))
+        assertEquals(null, parseGeometryPoint(null))
+    }
+
+    @Test
+    fun truncateDoubleTest() {
+        assertEquals("5", truncateDouble("5"))
+        assertEquals("-5", truncateDouble("-5"))
+        assertEquals("5.12", truncateDouble("5.12"))
+        assertEquals("-5.12", truncateDouble("-5.12"))
+        assertEquals("5.12", truncateDouble("5.1234"))
+        assertEquals("-5.12", truncateDouble("-5.1234"))
+        assertEquals("", truncateDouble(""))
+        assertEquals("", truncateDouble(null))
+        assertEquals("", truncateDouble("qwerty"))
+    }
+
+    @Test
+    fun parseGeometryTest() {
+        assertThat(parseGeometry("1.0 2.0 3 4"), equalTo(listOf(MapPoint(1.0, 2.0, 3.0, 4.0))))
+        assertThat(
+            parseGeometry("1.0 2.0 3 4; 5.0 6.0 7 8"),
+            equalTo(listOf(MapPoint(1.0, 2.0, 3.0, 4.0), MapPoint(5.0, 6.0, 7.0, 8.0)))
+        )
+
+        assertThat(parseGeometry("blah"), equalTo(emptyList()))
+        assertThat(parseGeometry("1.0 2.0 3 4; blah"), equalTo(emptyList()))
+        assertThat(
+            parseGeometry("37.45153333333334 -122.15539166666667 0.0 qwerty"),
+            equalTo(emptyList())
+        )
+    }
+
+    @Test
+    fun isWithinMapBoundsTest() {
+        assertThat(isWithinMapBounds(MapPoint(90.0, 0.0, 0.0, 0.0)), equalTo(true))
```

Comment:
It would be better to use `asserTreu` and `assertFalse` here.

Reply:
I'm not a big fan of one of those options but using `assertThat` in cases when we can use `assertTrue`/`assertFalse`/`assertNull` is a bit verbose to me. @lognaturel any thoughts?

Reply:
Ok here is a separate issue for that https://github.com/getodk/collect/issues/5436.